utils/test_matrix/test_matrix.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"golang.org/x/tools/go/packages"
+)
+
+const flowPackagePrefix = "github.com/onflow/flow-go/"
+const testMatrixFile = "test_matrix.json"
+const ciMatrixName = "dynamicMatrix"
+
+// testMatrix represents a single GitHub Actions test matrix combination that consists of a name and a list of flow-go packages associated with that name.
+type testMatrix struct {
+	Name     string `json:"name"`
+	Packages string `json:"packages"`
+}
+
+// Generates a list of packages to test that will be passed to GitHub Actions
+func main() {
+	if len(os.Args) == 1 {
+		fmt.Println("must have at least 1 package listed")
+		return
+	}
+
+	allFlowPackages := listAllFlowPackages()
+
+	targetPackages, seenPackages := listTargetPackages(os.Args[1:], allFlowPackages)
+
+	restPackages := listRestPackages(allFlowPackages, seenPackages)
+
+	testMatrix := generateTestMatrix(targetPackages, restPackages)
+
+	// generate JSON output that will be read in by CI matrix
+	// can't use json.MarshalIndent because fromJSON() in CI can’t read JSON with any spaces
+	testMatrixBytes, err := json.Marshal(testMatrix)
+	if err != nil {
+		panic(err)
+	}
+	err = os.WriteFile(testMatrixFile, testMatrixBytes, 0666)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
how do we use that file?

Reply:
I removed it in https://github.com/onflow/flow-go/pull/3087/commits/635fbd32ef269e53491cecce5c373fa809f5a83e